src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -560,11 +562,21 @@ export class NotebookService extends Disposable implements INotebookService {
 		if (notebookUri.scheme === Schemas.untitled) {
 			return true;
 		}
+		const trustedBooksConfigKey = 'notebook.trustedBooks';
 
 		let cacheInfo = this.trustedNotebooksMemento.trustedNotebooksCache[notebookUri.toString()];
 		if (!cacheInfo) {
-			// This notebook was never trusted
-			return false;
+			// Check if the notebook belongs to a book that's trusted
+			// and is not part of untrusted queue.
+			let trustedBookDirectories: string[] = !this._unTrustedCacheQueue.find(n => n === notebookUri) ? this._configurationService?.getValue(trustedBooksConfigKey) ?? [] : [];
```

Comment:
Why do we even have an "untrusted cache queue" (and why is it a "queue"? It's just a list...not a queue). Shouldn't anything that's not trusted therefore be untrusted?

Reply:
Not sure! @joberume can you explain?

Reply:
Introduction of untrusted queue is here: https://github.com/microsoft/azuredatastudio/commit/73b9849dd59714b08631defda762df556113de9c

From reading the code, it looks like it removes the notebook from the trustedCacheQueue.

Reply:
If we removed the ability to untrust a notebook, I certainly wouldn't complain; nobody ever signed off on the initial PR.

Oh wait, did I say that out loud 😄?

Reply:
@joberume oops my bad, I assumed the trust notebook code is entirely done by you. Should have checked the code! Thanks for chiming in anyway. :)